daffodil-runtime1/src/main/scala/org/apache/daffodil/dpath/FNFunctions.scala

```diff
@@ -803,6 +807,90 @@ case class FNFloor(recipe: CompiledDPath, argType: NodeInfo.Kind) extends FNOneA
   }
 }
 
+case class DFDLXLeftShift(recipes: List[CompiledDPath], argType: NodeInfo.Kind) extends FNTwoArgs(recipes)  {
+  override def computeValue(arg1: DataValuePrimitive, arg2: DataValuePrimitive, dstate: DState): DataValuePrimitive = {
+
+    val shiftLong = arg2.getLong
+    val shift = shiftLong.toInt
+    val width = argType.asInstanceOf[PrimNumeric].width.get
+    Assert.invariant(shift >= 0)
+    if(shift>=width)
+      dstate.SDE("Left shift not supported")
+
+    argType match {
+      case NodeInfo.Long => {
+         arg1.getLong << shift
+      }
+      case NodeInfo.Int => {
+         arg1.getInt << shift
```

Comment:
This will be cleaner if you just get them onto 1 line as in
 `case NodeInfo.Int => arg1.getInt << shift`
This spreading everything out on multiple lines with curly braces just dilutes code density and makes structural similarity harder to spot.